Rakefile

```diff
@@ -14,66 +14,48 @@ task :generate_csv do
-      name: r[:arearegion]
-    }
-
-    next unless r[:areasub_region]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region]),
-      name: r[:areasub_region]
-    }
-
-    next unless r[:areadistrict]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict]),
-      name: r[:areadistrict]
-    }
-
-    next unless r[:areaconstituency]
-    ocd_ids << {
-      id: id_for(region:r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict], constituency: r[:areaconstituency]),
-      name: r[:areaconstituency]
-    }
+  def ocd_ids_from_csv(csv_data, mapping = {})
+    csv = CSV.parse(csv_data, headers: true, header_converters: :symbol)
+    ocd_ids = Set.new
+
+    csv.each do |r|
+      region = r[mapping.fetch(:region, :region)]
+      next unless region
+      ocd_ids << {
+        id: id_for(region: region),
+        name: region.strip
+      }
+
+      subregion = r[mapping.fetch(:subregion, :subregion)]
+      next unless subregion
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion),
+        name: subregion.strip
+      }
+
+      district = r[mapping.fetch(:district, :district)]
+      next unless district
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion, district: district),
+        name: district.strip
+      }
+
+      constituency = r[mapping.fetch(:constituency, :constituency)]
+      next unless constituency
+      next if ['-', '_'].include?(constituency)
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion, district: district, constituency: constituency),
+        name: constituency.strip
+      }
+    end
+
+    ocd_ids
   end
 
-  area_csv = CSV.parse(open('area_info.csv').read, headers: true, header_converters: :symbol)
-
-  area_csv.each do |r|
-    next unless r[:region]
-    ocd_ids << {
-      id: id_for(region: r[:region]),
-      name: r[:region].strip
-    }
-
-    next unless r[:subregion]
-    ocd_ids << {
-      id: id_for(region: r[:region], subregion: r[:subregion]),
-      name: r[:subregion].strip
-    }
+  google_csv = open('https://docs.google.com/spreadsheets/d/1LetNFNq6ovg4bbq-Whze0Q06CYSpNqpRIzSUb9yVtw4/export?format=csv').read
+  area_csv = open('area_info.csv').read
 
-    next unless r[:district]
-    ocd_ids << {
-      id: id_for(region: r[:region], subregion: r[:subregion], district: r[:district]),
-      name: r[:district].strip
-    }
-
-    next unless r[:constituency]
-    next if ['-', '_'].include?(r[:constituency])
-    ocd_ids << {
-      id: id_for(region: r[:region], subregion: r[:subregion], district: r[:district], constituency: r[:constituency]),
-      name: r[:constituency].strip
-    }
-  end
+  ocd_ids = ocd_ids_from_csv(google_csv, region: :arearegion, subregion: :areasub_region, district: :areadistrict, constituency: :areaconstituency) + ocd_ids_from_csv(area_csv)
```

Comment:
This took a couple of readings to spot what was going on. I'd be tempted to make those mappings an explicit Hash to stand out more, rather than looking like this is taking a long list of arguments.